docs/en/development/tests.md

```diff
@@ -39,12 +39,40 @@ To add new test, create a `.sql` or `.sh` file in `queries/0_stateless` director
 
 Tests should use (create, drop, etc) only tables in `test` database that is assumed to be created beforehand; also tests can use temporary tables.
 
+### Restricting test runs
+
+A test can have one or more _test tags_ specifying restrictions for test runs.
+
+For `.sh` tests tags are placed in `.sh` files as comments, example:
+
+```bash
+#!/usr/bin/env bash
+# Tags: no-fasttest
+```
+
+For `.sql` tests tags are placed as SQL comments, example: 
+
+```sql
+-- Tags: no-fasttest
+SELECT 1
+```
+
+|Tag name | What it does | Usage example | Test example |
+|---|---|---|---|
+| `no-fasttest`|  Test is not run under [Fasttest](continuous-integration#fast-test) | Test uses MySQL table engine which is disabled in Fasttest| |
+| `no-zookeeper` |   |   | |
+| `no-long` |   |   | |
+| `no-replicated-database` |   |   | |
+| `no-parallel` |   |   | |
+| `distributed` |   |   | |
+| `no-random-settings` | | | |
+| `no-backward-compatibility-check` | | |  |
```

Comment:
Also we can specify particular features that are required for test. (In our CI it can affect just fasttest build that is compiled without some libraries). Flags are from `system.build_options`. Example `use-rocksdb`

https://github.com/ClickHouse/ClickHouse/pull/33780